frontend-react/src/components/MessageTracker/MessageDetails.tsx

```diff
@@ -15,57 +18,157 @@ type MessageDetailsProps = {
     id: string | undefined;
 };
 
+const parseFileLocation = (
+    fileLocation: string
+): {
+    fileLocation: string;
+    sendingOrg: string;
+    fileName: string;
+} => {
+    let fileLocationFromStr = "";
+    let sendingOrgFromStr = "";
+    let fileNameFromStr = "";
+
+    let fileReportsLocation = fileLocation.split("reports/").pop();
+    if (fileLocation.includes("reports/") && fileReportsLocation) {
+        fileLocationFromStr = fileReportsLocation.split("%2F")[0].toUpperCase();
```

Comment:
We're creating a new array on each invocation of `split` here -- we can just keep the one reference around and pick from it:

```tsx
[fileLocation, sendingOrg, fileName] = fileReportsLocation.split("%2F");
```

Or if we want to make all this logic a one-liner:

```tsx
[fileLocation, sendingOrg, fileName] = fileLocation.split("/").slice(-1)[0].split("%2F");
```

Regardless, let's wrap this in a `try-catch` and return empty string just in case there's a format issue, though!